pkg/operation/botanist/component/extensions/worker/worker.go

```diff
@@ -353,3 +377,24 @@ func (w *worker) findNodeTemplateAndMachineTypeByPoolName(ctx context.Context, o
 	}
 	return nil, ""
 }
+
+// CheckWorkerStatusMachineDeployments checks if the status of the worker is updated or not during its reconciliation.
+// It is updated if
+// * The status.MachineDeploymentsLastUpdateTime > the value of the time stamp stored in worker struct before the reconciliation begins.
+func (w *worker) checkWorkerStatusMachineDeployments(o client.Object) error {
```

Comment:
```suggestion
func (w *worker) checkWorkerStatusMachineDeploymentsUpdated(o client.Object) error {
```

Reply:
addressed in [14b7b28](https://github.com/gardener/gardener/pull/7774/commits/14b7b281d57cfd42529151c448f8c0d84ba4f68b)